apps/frontend/app/(client)/(main)/settings/page.tsx

```diff
@@ -38,31 +39,27 @@ export default function Page() {
   const updateNow = searchParams.get('updateNow')
   const router = useRouter()
   const bypassConfirmation = useRef<boolean>(false)
-  const [defaultProfileValues, setdefaultProfileValues] = useState<Profile>({
-    username: '',
-    userProfile: {
-      realName: ''
+
+  const { data: defaultProfileValues, isLoading } = useQuery({
+    ...profileQueries.fetch(),
+    initialData: {
+      username: '',
+      userProfile: {
+        realName: ''
+      },
+      studentId: '',
+      major: ''
     },
-    studentId: '',
-    major: ''
+    retry: false
   })
 
-  // Fetch default profile values
+  const [majorValue, setMajorValue] = useState(defaultProfileValues.major)
+
   useEffect(() => {
-    const fetchDefaultProfile = async () => {
-      try {
-        const data: Profile = await safeFetcherWithAuth.get('user').json()
-        setMajorValue(data.major)
-        setdefaultProfileValues(data)
-        setIsLoading(false)
-      } catch (error) {
-        console.error('Failed to fetch profile:', error)
-        toast.error('Failed to load profile data')
-        setIsLoading(false)
-      }
+    if (defaultProfileValues.major) {
+      setMajorValue(defaultProfileValues.major)
     }
-    fetchDefaultProfile()
-  }, [])
+  }, [defaultProfileValues.major, isLoading])
```

Comment:
```suggestion
  }, [defaultProfileValues.major])
```

여기도 `isLoading`은 빼도 될 것 같아요!